features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SystemRequirementConfiguration.kt

```diff
@@ -0,0 +1,113 @@
+        if (systemRequirement == null) {
+            systemRequirement = DefaultSystemRequirement()
+        }
+        systemRequirement?.runIfMeetRequirements(
+            systemRequirementConfiguration = this,
+            checkers = listOf(
+                CpuRequirementChecker(internalLogger = internalLogger),
+                MemoryRequirementChecker(internalLogger = internalLogger)
+            ),
+            internalLogger = internalLogger,
+            runnable = runnable
+        )
+    }
+
+    /**
+     * Builder class for configuring and creating instances of the [SystemRequirementConfiguration] needed to
+     * enable the Session Replay feature.
+     *
+     */
+    class Builder {
+        private var minCPUCoreNumber: Int = 0
+        private var minRAMSize: Int = 0
+
+        /**
+         * Sets the minimum CPU core number requirement.
+         *
+         * @param cpuCoreNumber The minimum CPU core number.
+         * @return The current Builder instance for chaining.
+         */
+        fun setMinCPUCoreNumber(cpuCoreNumber: Int): Builder {
+            this.minCPUCoreNumber = cpuCoreNumber
+            return this
+        }
+
+        /**
+         * Sets the minimum RAM in megabytes.
+         *
+         * @param minRAMSize The minimum RAM in megabytes.
+         * @return The current Builder instance for chaining.
+         */
+        fun setMinRAM(minRAMSize: Int): Builder {
+            this.minRAMSize = minRAMSize
+            return this
+        }
+
+        /**
+         * Builds and returns an instance of the [SystemRequirementConfiguration] with the configured parameters.
+         *
+         * @return A fully configured instance of [SystemRequirementConfiguration].
+         */
+        fun build(): SystemRequirementConfiguration {
+            return SystemRequirementConfiguration(
+                minCPUCoreNumber = minCPUCoreNumber,
+                minRAMSize = minRAMSize
+            )
+        }
+    }
+
+    companion object {
+
+        /**
+         * A preconfigured instance representing the default system requirements for enabling the Session Replay feature.
+         *
+         * This instance provides reasonable default values for most systems:
+         * - Minimum RAM: 1024 MB
+         * - Minimum CPU core number: 2
+         *
+         * Use this instance when standard system requirements are sufficient.
+         */
+        val DEFAULT: SystemRequirementConfiguration = Builder()
+            .setMinRAM(1024)
+            .setMinCPUCoreNumber(2)
+            .build()
+
+        /**
+         * A special instance representing no system requirements for enabling the Session Replay feature.
+         *
+         * With this instance, Session Replay will be enabled regardless of system specifications.
+         * This is useful in cases where system requirements should not restrict functionality.
+         */
+        val NONE: SystemRequirementConfiguration = Builder().build()
```

Comment:
minor: I think you can simplify the things here a bit and just call `SystemRequirementConfiguration` constructor here directly.